toucan_connectors/snowflake_common.py

```diff
@@ -142,6 +142,19 @@ def render_datasource(self, datasource: SfDataSource) -> dict:
             'parameters': prepared_query_parameters,
         }
 
+    def add_default_order_if_needed(
+        self,
+        query: str,
+        query_columns: Optional[List] = None,
+    ):
+        """If it's a SELECT type query and there is not an 'ORDER BY' inside, we sort using ROW_NUMBER function"""
+
+        if query_columns and 'WITH SELECT_STEP_' in query and ' ORDER BY ' not in query:
```

Comment:
This may be sensitive to case, where SQL operators should not.
What do you think about doing a case *in*sensistive search? I would use a regexp (`re`) for this task.

Reply:
the `re` search in this case may be too expensive in computing for each query sent to snowflake... is it not better to just uppercase the query and do checks ? 🤔 
```
 if query_columns and 'WITH SELECT_STEP_' in query.upper() and ' ORDER BY ' not in query.upper():
 ```
 
 What do you think ?

Reply:
I don't know how expensive would be a regex search, but yes, using upper is already a good idea!
There also could be issues with the spaces before or after (could be line beeaks? or maybe other characters?), but that's very edgy.
So let's go for the upper! :+1:

Reply:
You marked a point for the line breaks between ORDER and BY so finally i used the regex search on that !